### ISSUE TITLE
[BUG] ANSIString.split() breaks with MXP

### ISSUE THREAD

**Issue by InspectorCaracal** [evennia/evennia#3110]
#### Describe the bug
The current implementation of ANSIString has a property `_clean_string` which it uses for things like splitting, which involves the ANSI flags having been converted to ANSI codes as part of turning into ANSIString, and then stripped for the clean string.

However, the MXP codes are not part of this process at all, so the `_clean_string` property contains the MXP flags and embedded command. This works okay in most situations, since the string is ultimately parsed further down the line to regular MXP tags or stripped, EXCEPT when you try split the ANSIString.

If your separator character is in the command - for example, a space - then it will split the command string _as well as_ the actual display text. If it's just concatenated back together, it works fine - but if it's being wrapped, for example, and the command text is longer than the wrap width, then it breaks apart the command and/or display string in strange ways.

#### To Reproduce
```python
ansi_test = ANSIString("|lctest command|ltthis is a test|le")
logger.log_msg(ansi_test.split())
```

#### Expected behavior
The "invisible" portion of the MXP, i.e. the command string, should not be split up

#### Environment, Evennia version, OS etc
evennia 1.1.1

#### Additional context
This appears to be the underlying cause of both #3082 and #3027 

It's really not clear what the best approach for solving this issue is. None of the existing ANSIString logic for parsing around the ANSI codes themselves apply, since the MXP structure is tag-based instead of switch-based. It's also not clear what change caused this to start happening in the first place.

**Comment by InspectorCaracal** [comment]
The main issue I see for solving this problem fully is that even if you only split the display text itself, the "chunks" will no longer be able to correctly parse out the MXP. `[ "|lctest", "command|ltthis", "is", "a", "test|le" ]` obviously won't work, but `[ "|lctest command|ltthis", "is", "a", "test|le" ]` won't either - the `|lc` `|lt` and `|le` tags aren't identified as mxp when isolated in separate strings. You could theoretically parse it into `[ "|lc", "test command", "|lt", "this", "is", "a", "test", "|le" ]` and do additional manipulation on that, but that starts requiring a lot of special case checks.

**Comment by Griatch** [comment]
Yeah, this is not likely to be something we will address until we move towards client-side rendering (where ANSIString will not be a thing on the server side at all).
